packages/admin-ui/src/constants/message_severity.ts

```diff
@@ -0,0 +1,6 @@
+export enum MESSAGE_SEVERITY {
```

Comment:
I do like this, making impossible states not allowed

Reply:
Is there any way to split it up somehow into logical pieces?

That was always the plan ... I was also hoping  to have it as a reference if only to ease conversations where I can point at aspects of it - use it as a short hand for want we want to do ... or more likely - given new information say "this is what is looks like when I go wrong."

When I start make a list of obvious fragmentation ... These float to the top... which I intend to work on.

A) MESSAGE_SEVERITY -- issue

B) The ACTION_TYPE issue 
  ... although what I have done is moved CONSTANTS that were contained in separate modules and moved them into a single file  constants/action_type.ts  [ I have bunched the constants so that I keep the history of which file I moved them from and left comments like "see roles.ts"  ] But I struggle with this design choice ... should I keep the constant in the relevant module and make a mega ENUM using the union operator '|' .. Should I be using interface not ENUM ... I am still sitting on that. If they have to be unique and do not need to be iterable .. well that is the definition of a SET over MAP in javascript. hmm

C) The Node/Schema/Content issue....
     utils/api.ts   Has a "Node" interface ... which I think is a bodge. Hand crafted typescript which has to be keep in sync with changes to core.Agh  There are similarly troubling Schema and Content interfaces. But hey this is a rough draft. The correct thing todo is to autogenerate this for a core definition somehow...

D) FileUpload and FileUploadWidget... have too complex a set of data structures. I appreciate to some extent we have to receive radically different structures base on whether we are in single or multiple mode... but I am looking for a more elegant expression of that fixed.complexity.

E) Slowly convert each sub_atom and then widget in is small easy to review PR's

Reply:
B) It looks like you can iterate over ENUM's, see https://github.com/Microsoft/TypeScript/issues/17198#issuecomment-315400819
C) It feels like this could be its own issue where we first discuss how a good solution could look like, instead of trying to make a PR directly. What you describe as autogeneration might be tricky though. When the content model changes, we can't actually change our types, I guess.
D) I think FileUpload/FileUploadWidget might be up for some general refactoring. As part of that we could introduce typescript, as it would help to do that
E) Slowly convert each sub_atom and then widget in is small easy to review PR's